### ISSUE TITLE
AnalyticalTable: A11Y - Table update and no data announcements missing for screen readers

### ISSUE THREAD

**Issue by fushizen** [SAP/ui5-webcomponents-react#7282]
### Describe the bug

Using JAWS 2025, the AnalyticalTable component has a few A11Y issues:

1. On table data update / refresh, there is no announcement for screen readers that the table data has been updated
2. On using the default no data text, there is no announcement that the table has no data (the focus never moves to the `No data` text area, screenshot attached, same behaviour for issue 3. below)
3. On using a custom `IllustratedMessage` as the no data component, it is not possible to focus on / move the control to the no data component, thus the component is never announced by the screen reader i.e. there is no announcement that the table has no data when using a custom `NoDataComponent`

### Isolated Example

https://stackblitz.com/edit/github-3ajh2g-peem6xxa?file=src%2FApp.tsx

### Reproduction steps

1. Open JAWS
2. Go to the [storybook for AnalyticalTable](https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/data-display-analyticaltable--docs) / linked StackBlitz
3. There are no table data update announcements or no data announcements

### Expected Behaviour

Screen reader announcements for when table data is updated and when there is no data in the table / custom no data component is used

### Screenshots or Videos

![Image](https://github.com/user-attachments/assets/3587280c-b326-4cf3-ab69-3bef085411e6)

### UI5 Web Components for React Version

2.9.0

### UI5 Web Components Version

2.9.0

### Browser

Edge

### Relevant log output

```shell

```

### Organization

SAP

### Declaration

- [x] I’m not disclosing any internal or sensitive information.
